src/accel/RunAction.cc

```diff
@@ -59,12 +83,122 @@ void RunAction::EndOfRunAction(const G4Run*) {}
 void RunAction::build_core_params()
 {
     G4AutoLock lock(&mutex);
-    if (false)
+    if (params_->params)
     {
         // Some other thread constructed params between the thread-unsafe check
         // and this thread-safe check
         return;
     }
+
+    celeritas::GeantImporter load_geant_data(
+        G4TransportationManager::GetTransportationManager()
+            ->GetNavigatorForTracking()
+            ->GetWorldVolume());
+
+    auto imported = load_geant_data();
+    CELER_LOG_LOCAL(info) << "loaded data: "
+                          << (imported ? "success" : "failure");
+
+    CoreParams::Input params;
+
+    // Create action manager
+    {
+        params.action_reg = std::make_shared<ActionRegistry>();
+    }
+
+    // Load geometry
+    {
+        params.geometry = std::make_shared<GeoParams>(options_->geometry_file);
+        if (!params.geometry->supports_safety())
+        {
+            CELER_LOG(warning)
+                << "Geometry contains surfaces that are "
+                   "incompatible with the current ORANGE simple "
+                   "safety algorithm: multiple scattering may "
+                   "result in arbitrarily small steps";
+        }
```

Comment:
Since we're can't really use ORANGE for Acceleritas, we can probably delete this warning.